agent/engine/docker_task_engine.go

```diff
@@ -329,6 +362,16 @@ func (engine *DockerTaskEngine) synchronizeContainerStatus(container *api.Docker
 		// update the container known status
 		container.Container.SetKnownStatus(currentState)
 	}
+	// Update task ExecutionStoppedAt timestamp
+	if container.Container.GetKnownStatus() == api.ContainerStopped && container.Container.IsEssential() {
+		now := engine.time().Now()
+		ok := task.SetExecutionStoppedAt(now)
+		if ok {
+			seelog.Infof("Recording execution stopped time for a task, essential container in task stopped, task %s, time: %s",
```

Comment:
Once #1208 is merged, please rebase this on top of that and don't forget to update the log string as per the format being followed in that:
* `"Recording execution ` -> `"Task engine [%s] Recording execution `
* instead of `task.String()`, use `task.Arn`